node/forest_libp2p/src/rpc/cbor_codec.rs

```diff
@@ -111,39 +80,29 @@ impl<'de, Item: Deserialize<'de>> Default for Decoder<Item> {
     }
 }
 
-impl<'de, Item: Deserialize<'de>> IoDecoder for Decoder<Item> {
+impl<Item: Serialize + DeserializeOwned> IoDecoder for Decoder<Item> {
     type Item = Item;
     type Error = Error;
 
     fn decode(&mut self, src: &mut BytesMut) -> Result<Option<Item>, Error> {
-        let mut pos = 0;
-        let mut slice: &[u8] = src;
-
-        let reader = IoRead::new(Counted {
-            r: &mut slice,
-            pos: &mut pos,
-        });
+        // let mut pos = 0;
+        let slice: &[u8] = src;
+        let reader = BufReader::new(slice);
 
         // Use the deserializer directly, instead of using `deserialize_from`. We explicitly do
         // *not* want to check that there are no trailing bytes ‒ there may be, and they are
         // the next frame.
         //
-        let mut deserializer = Deserializer::new(reader);
-
-        match Item::deserialize(&mut deserializer) {
+        // let mut deserializer = Deserializer::new(reader);
+        match serde_ipld_dagcbor::de::from_reader_unstrict(reader) {
             // If we read the item, we also need to consume the corresponding bytes.
-            //
             Ok(item) => {
-                src.advance(pos);
+                let offset = fvm_ipld_encoding::to_vec(&item).expect("Infallible").len();
```

Comment:
counted wrapper does not work here either because `from_reader` now takes a `BufReader`

Reply:
nvm this file has been removed